### ISSUE TITLE
训练时报错显示显存不足Check failed: ptr Fail to allocate GPU memory 658816 bytes 

### ISSUE THREAD

**Issue by bit1002lst** [PaddlePaddle/Paddle#6440]
但是调用时已经把参数设置为
./paddle cluster_train \
  --config test/api_train_v2.py \
  --port 12864 \
  --use_gpu cpu \

**Comment by Canpio** [comment]
`use_gpu·这个参数的作用是为python脚本提供一个环境变量，python脚本中可以通过读取这个环境变量来决定是使用CPU还是GPU。如果已经在脚本中写死使用GPU，那么在这里设置是没有用的。请检查Python脚本中是否写死了用GPU。

**Comment by bit1002lst** [comment]
刚才检查了一下，python中关于gpu的设置是通过读取命令行进行配置的，代码如下：
`import math
import os
import paddle.v2 as paddle
import pickle

embsize = 32
hiddensize = 256
N = 5
cluster_train_file = "./train_data_dir/train/train.txt"
cluster_test_file = "./test_data_dir/test/test.txt"
node_id = os.getenv("OMPI_COMM_WORLD_RANK")
if not node_id:
    raise EnvironmentError("must provied OMPI_COMM_WORLD_RANK")

def wordemb(inlayer):
    wordemb = paddle.layer.embedding(
        input=inlayer,
        size=embsize,
        param_attr=paddle.attr.Param(
            name="_proj",
            initial_std=0.001,
            learning_rate=1,
            l2_rate=0,
            sparse_update=True))
    return wordemb

def cluster_reader_cluster(dirname, node_id):
    def cluster_reader():
        for fn in os.listdir(dirname):
            with open(dirname+"/"+fn, "r") as f:
                for l in f:
                    csv_data = [int(cell) for cell in l.split(",")]
                    yield tuple(csv_data)
    return cluster_reader

def main():
    # get arguments from env

    # for local training
    TRUTH = ["true", "True", "TRUE", "1", "yes", "Yes", "YES"]
    cluster_train = os.getenv('PADDLE_CLUSTER_TRAIN', "False") in TRUTH
    use_gpu = os.getenv('PADDLE_USE_GPU', "False")

    if not cluster_train:
        paddle.init(use_gpu=use_gpu, trainer_count=int(os.getenv("PADDLE_TRAINER_COUNT", "1")))
    else:
        paddle.init(
            use_gpu=use_gpu,
            trainer_count=int(os.getenv("PADDLE_TRAINER_COUNT", "1")),
            port=int(os.getenv("PADDLE_PORT", "7164")),
            ports_num=int(os.getenv("PADDLE_PORTS_NUM", "1")),
            ports_num_for_sparse=int(os.getenv("PADDLE_PORTS_NUM_FOR_SPARSE", "1")),
            num_gradient_servers=int(os.getenv("PADDLE_NUM_GRADIENT_SERVERS", "1")),
            trainer_id=int(os.getenv("PADDLE_TRAINER_ID", "0")),
            pservers=os.getenv("PADDLE_PSERVERS", "127.0.0.1"))
    #word_dict = paddle.dataset.imikolov.build_dict()
    fn = open("thirdparty/thirdparty/word_dict.pickle", "r")
    word_dict = pickle.load(fn)
    fn.close()
    dict_size = len(word_dict)
    firstword = paddle.layer.data(
        name="firstw", type=paddle.data_type.integer_value(dict_size))
    secondword = paddle.layer.data(
        name="secondw", type=paddle.data_type.integer_value(dict_size))
    thirdword = paddle.layer.data(
        name="thirdw", type=paddle.data_type.integer_value(dict_size))
    fourthword = paddle.layer.data(
        name="fourthw", type=paddle.data_type.integer_value(dict_size))
    nextword = paddle.layer.data(
        name="fifthw", type=paddle.data_type.integer_value(dict_size))

    Efirst = wordemb(firstword)
    Esecond = wordemb(secondword)
    Ethird = wordemb(thirdword)
    Efourth = wordemb(fourthword)

    contextemb = paddle.layer.concat(input=[Efirst, Esecond, Ethird, Efourth])
    hidden1 = paddle.layer.fc(input=contextemb,
                              size=hiddensize,
                              act=paddle.activation.Sigmoid(),
                              layer_attr=paddle.attr.Extra(drop_rate=0.5),
                              bias_attr=paddle.attr.Param(learning_rate=2),
                              param_attr=paddle.attr.Param(
                                  initial_std=1. / math.sqrt(embsize * 8),
                                  learning_rate=1))
    predictword = paddle.layer.fc(input=hidden1,
                                  size=dict_size,
                                  bias_attr=paddle.attr.Param(learning_rate=2),
                                  act=paddle.activation.Softmax())

    def event_handler(event):
        if isinstance(event, paddle.event.EndIteration):
            if event.batch_id % 100 == 0:
                result = trainer.test(
                    paddle.batch(
                        cluster_reader_cluster("./test_data_dir/test/", node_id), 32))
                print "Pass %d, Batch %d, Cost %f, %s, Testing metrics %s" % (
                    event.pass_id, event.batch_id, event.cost, event.metrics,
                    result.metrics)

    cost = paddle.layer.classification_cost(input=predictword, label=nextword)
    parameters = paddle.parameters.create(cost)
    adagrad = paddle.optimizer.AdaGrad(
        learning_rate=3e-3,
        regularization=paddle.optimizer.L2Regularization(8e-4))
    trainer = paddle.trainer.SGD(cost,
                                 parameters,
                                 adagrad,
                                 is_local=not cluster_train)
    trainer.train(
        paddle.batch(cluster_reader_cluster("./train_data_dir/train/", node_id), 32),
        num_passes=30,
        event_handler=event_handler)


if __name__ == '__main__':
    main()

`

**Comment by Canpio** [comment]
issue中给出的是否是完整的提交命令？为什么最后一行有个换行而前面没有呢？

**Comment by typhoonzero** [comment]
`use_gpu = os.getenv('PADDLE_USE_GPU', "False")` 这样默认将获得一个string而不是bool变量。

**Comment by typhoonzero** [comment]
Closing due to low activity.
